app/controllers/projects/group_links_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Projects
+  # Controller actions for Project Group Links
+  class GroupLinksController < Projects::ApplicationController
+    include ShareActions
+
+    def group_link_params
+      params.require(:namespace_group_link).permit(:id, :group_id, :namespace_id, :group_access_level, :expires_at)
+    end
+
+    private
+
+    def namespace_group_link
+      @namespace_group_link ||= NamespaceGroupLink.find_by(id: params[:id])
```

Comment:
**nitpick (blocking):** We should probably change this query to be scoped to the `Project`, so that might look like `@project.namespace.shared_with_group_links.find_by(id: params[:id])`

The idea being here, that we don't want to allow editing a NamespaceGroupLink in a place where it should not be editted.

Reply:
Updated in 07cfcddc1c8ffc87c49116e40478b27068c0ee56